src/ocean/global.F90

```diff
@@ -251,53 +248,85 @@ integer function find_row2(i,j,k,XX)
 
   end function find_row2
 
-  !! this is a part of the subroutine grid from grid.f. We need it to
-  !! initialize the vectors x,y,z...
-  !! It should be kept consistent with that subroutine
-  SUBROUTINE g_grid
-    use m_par
+  subroutine get_x(asize,ptr) bind(C,name="get_global_x")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(x)
+    ptr = c_loc(x(1))
+  end subroutine get_x
+
+  subroutine get_xu(asize,ptr) bind(C,name="get_global_xu")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(xu)
+    ptr = c_loc(xu(0))
+  end subroutine get_xu
+
+  subroutine get_y(asize,ptr) bind(C,name="get_global_y")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(y)
+    ptr = c_loc(y(1))
+  end subroutine get_y
+
+  subroutine get_yv(asize,ptr) bind(C,name="get_global_yv")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(yv)
+    ptr = c_loc(yv(0))
+  end subroutine get_yv
+
+  subroutine get_z(asize,ptr) bind(C,name="get_global_z")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(z)
+    ptr = c_loc(z(1))
+  end subroutine get_z
+
+  subroutine get_ze(asize,ptr) bind(C,name="get_global_ze")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(ze)
+    ptr = c_loc(ze(1))
+  end subroutine get_ze
+
+  subroutine get_zw(asize,ptr) bind(C,name="get_global_zw")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(zw)
+    ptr = c_loc(zw(0))
+  end subroutine get_zw
+
+  subroutine get_zwe(asize,ptr) bind(C,name="get_global_zwe")
+    use, intrinsic :: iso_c_binding
     implicit none
-    integer i,j,k
-    !     EXTERNAL
-    real  fz
-
-    write(*,*) '============GRID==========='
-    write(*,10) xmin*180/pi,xmax*180/pi,ymin*180/pi,ymax*180/pi
-10  format(1x,'configuration:[',f6.1,1x,f6.1,'] x [',f6.1,1x,f6.1,']')
-    write(*,20) n,m,l
-20  format(1x,'resolution:',i8,'x',i8,'x',i8)
-    write(*,30) qz
-30  format(1x,'stretching parameter',g12.4)
-    write(*,*) '============GRID==========='
-
-    dx = (xmax-xmin)/N
```

Comment:
These variables now seems to be uninitialized in m_global? Please remove all of the unused variables.

Reply:
Good catch, these are still used in levitus, so I've kept them in the new version, unless @erik808 decides we can/should delete `levitus.F90`.